test-data/unit/lib-stub/dataclasses.pyi

```diff
@@ -32,3 +32,5 @@ def field(*,
 
 
 class Field(Generic[_T]): pass
+
+def replace(obj: _T, **changes: Any) -> _T: ...
```

Comment:
The stub has:
```
def replace(__obj: _DataclassT, **changes: Any) -> _DataclassT: ...
```
I can rename `obj` to `__obj` in this fixture. We can also change the stub to better reflect [reality](https://github.com/python/cpython/issues/81344):
```python
if sys.version_info >= (3, 9):
    def replace(obj: _DataclassT, /, **changes: Any) -> _DataclassT: ...
else:
   def replace(obj: _DataclassT, **changes: Any) -> _DataclassT: ...
```
Can we just use the Python 3.9 syntax even for Python 3.8 targets? Passing `obj` as kw is deprecated in Python 3.8 so being a checker/linter we might as well just disallow it, no?

Do you think it matters much? For the plugin to function, `replace` has to:
1. exist
2. have a first argument

The `**changes` are optional, the retval is overwritten, and FWIW it doesn't look for a keyword arg `obj` since it's deprecated since Python 3.8 and was never a good idea to begin with, so we might as well just not support it, right?

Reply:
> We can also change the stub to better reflect [reality](https://github.com/python/cpython/issues/81344):

No need: prepending the parameter name with `__` is the py37-compatible way of declaring that a parameter is positional-only, as is described in PEP 484: https://peps.python.org/pep-0484/#positional-only-arguments

I don't think you need to change anything in your existing fixtures/tests, I'm just suggesting adding a single additional test in the `pythoneval.test` file. Unlike mypy's other tests, the tests in that file don't use fixtures — they run with typeshed's full stdlib stubs. So it's a good sanity check to make sure that the machinery you've added works with typeshed's stubs for this function as well as with mypy's fixtures.

Reply:
7b907cf50fbadf873007d455d9fd301119a09872